contacts/main.js

```diff
@@ -0,0 +1,253 @@
+    finalArray.push(char);
+  }
+
+  let arr = '';
+  for (let i=0;i<finalArray.length; i++){
+    arr += finalArray[i];
+  }
+
+  return arr;
+}
+
+function addToHTMLNewPerson(newPerson) {
+  let ul = document.getElementsByClassName('contacts')[0].getElementsByTagName('ul')[0];
+
+  let newLi = document.createElement('li');
+  newLi.className = 'person-field';
+  newLi.id = 'li-' + newPerson.id;
+
+  let plusButton = document.createElement('button');
+  plusButton.id = 'plus-button-' + newPerson.id;
+  plusButton.innerHTML = '+';
+  newLi.appendChild(plusButton);
+
+  let fav_count = document.createElement('div');
+  fav_count.id = 'count-' + newPerson.id;
+  fav_count.className = 'favs-count'
+  fav_count.innerHTML = newPerson.likes;
+  newLi.appendChild(fav_count);
+
+  let minusButton = document.createElement('button');
+  minusButton.id = 'minus-button-' + newPerson.id;
+  minusButton.innerHTML = '-';
+  newLi.appendChild(minusButton);
+
+  let image = document.createElement('img');
+  image.src = newPerson.image;
+  image.alt = 'pic';
+  image.className = 'avatar';
+  newLi.appendChild(image);
+
+  let checkbox = document.createElement('input');
+  checkbox.type = 'checkbox';
+  newLi.appendChild(checkbox);
+
+  let full_name = document.createElement('span');
+  full_name.className = 'full-name';
+  let first_name = document.createElement('span');
+  first_name.innerHTML = newPerson.firstName;
+  first_name.style.marginRight = '10px';
+  full_name.appendChild(first_name);
+  let last_name = document.createElement('span');
+  last_name.innerHTML = newPerson.lastName;
+  full_name.appendChild(last_name);
+  newLi.appendChild(full_name);
+
+  let trash = document.createElement('img');
+  trash.src = 'images/trash-1.png';
+  trash.alt = 'trash';
+  trash.className = 'trash';
+  newLi.appendChild(trash);
+
+  ul.appendChild(newLi);
+}
+
+function attachConections(newPerson){
+  let personHTMLEntity = document.getElementById('li-'+newPerson.id);
+  // for trash
+  personHTMLEntity.getElementsByClassName('trash')[0].onclick = () => {
+    personHTMLEntity.remove();
+    //refresh the favs list
+  }
+  document.getElementById('plus-button-' + newPerson.id).onclick = () => {
+    newPerson.likes+=1;
+    document.getElementById('count-'+newPerson.id).innerHTML = newPerson.likes;
+    //refresh the favs list
+  }
+  document.getElementById('minus-button-' + newPerson.id).onclick = () => {
+    if (newPerson.likes > 0) {
+      newPerson.likes-=1;
+      document.getElementById('count-'+newPerson.id).innerHTML = newPerson.likes;
+    }
+    //refresh the favs list
+  }
+}
+
+function addNewPerson() {
+  // get the data from the form
+  let id = generateUniqueID();
+  let firstName = getFirstNameInput();
+  let lastName = getLastNameInput();
+  let image = getImageProfileInput();
+  // create a new person
+  let newPerson = new Person(id, firstName, lastName, image);
+  console.log( 'created -> ' + newPerson.getFullName());
```

Comment:
As a future note, it is not recommended to leave console.log statements inside your code.

Reply:
that was there for debugging